CMakeLists.txt

```diff
@@ -745,3 +773,46 @@ if (EMSCRIPTEN)
     LINK_DEPENDS "${WABT_POST_JS};${EMSCRIPTEN_EXPORTED_JSON}"
   )
 endif ()
+
+# Create find_package configuration files
+if (WABT_INSTALL_RULES)
+  include(CMakePackageConfigHelpers)
+
+  set(WABT_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/wabt"
+      CACHE STRING "Path to wabt CMake files")
+
+  install(
+    EXPORT wabt-targets
+    DESTINATION "${WABT_INSTALL_CMAKEDIR}"
+    NAMESPACE wabt::
```

Comment:
Is this reason we require the aliases above?   Are the aliases actually required?

Reply:
Linking in CMake is... fraught. 

Normally, when you link to a library `A` via `target_link_libraries(tgt PRIVATE A)`, CMake first checks to see if `A` is the name of a CMake target. If so, it can be smarter about how it links to `tgt`. However, if it can't find the target `A` then it shrugs its shoulders and says "eh, this is a link flag". In the best case, this defers what should be a configure-time error to a build-time linker failure. In the worst case, this _succeeds_ and links to an unexpected copy of `A` laying around in the user's system. Nasal demons, ahoy!

To fix this, CMake adopted the convention that namespaced (with `::`) targets are _always_ considered CMake targets and will never be silently treated as linker flags. If the target doesn't exist, you get an error at configure time. Unfortunately, there's a caveat: only `ALIAS` and `IMPORTED` targets can have a `:` in the name. Why? Because the `:` would conflict with drive letter syntax on Windows. So:

1. We _have_ to declare the build target `wabt` without a namespace.
2. But linking to non-namespaced targets invites bad behavior.
3. So we export it with a namespace as `wabt::wabt`. Good.
4. But now the `FetchContent` users have to link to `wabt` and if they want to switch to `find_package`, they have to change `wabt` to `wabt::wabt` elsewhere in their code. Bad.
5. So we create an alias in the build to match the exported target.

Now, both `FetchContent` and `find_package` users can be confident linking to `wabt::wabt`.